lib/vtul/dataingest.rb

```diff
@@ -0,0 +1,184 @@
+require 'csv'
+require 'digest'
+require 'fileutils'
+require 'json'
+require 'net/http'
+require 'uri'
+require 'yaml'
+
+class DataIngest
+    
+  def createreport(filename, content)
+    File.open(filename, 'w') { |file| file.write(content) }
+  end
+
+
+  def getfilelist(folderpath)
+    return Dir[folderpath].select{ |filename| File.file? filename }.map{ |filename| File.basename filename }
+  end
+
+
+  def getfileprefix(filename)
+    if filename.start_with?("dmf_")
+      return "dmf_"
+    elsif filename.start_with?("cgit_")
+      return "cgit_"
+    else
+      return ""
+    end
+  end
+
+
+  def geojson(content)
+    rec = {}
+    uuid = Digest::SHA256.hexdigest content[1] 
+    timestring = Time.now().strftime("%Y-%m-%dT%H:%M:%SZ") 
+
+    rec['uuid'] = uuid
+    rec['dc_identifier_s'] = content[1]
+    rec['dc_rights_s'] = content[2]
+    rec['dct_provenance_s'] = content[3]
+    if !content[4].nil?
+      rec['dct_references_s'] = "{\"http://schema.org/downloadUrl\":\"" + content[1] + "\"}"
+    else
+      rec['dct_references_s'] = "{\"http://schema.org/downloadUrl\":\"" + content[1] + "\",\"http://www.opengis.net/def/serviceType/ogc/wcs\":\"" + content[4] + "\"}"     
+    end
+    rec['dc_creator_sm'] = content[5]
+    rec['dc_language_s'] = content[6]
+    rec['dc_publisher_s'] = content[7]
+    rec['dc_type_s'] = content[9]
+    rec['dct_spatial_sm'] = content[10]
+    rec['dct_temporal_sm'] = content[11]
+    rec['dct_issued_s'] = timestring
+    rec['dct_isPartOf_sm'] = content[13]
+    rec['georss_box_s'] = content[14]
+    if !content[14].nil?
+      gdata = content[14].split(",")  
+      rec['solr_geom'] = "ENVELOPE("+ gdata[1] + "," + gdata[3] + "," + gdata[2] + "," + gdata[0] + ")"
+    end
+    rec['dc_title_s'] = content[17]
+    rec['dc_description_s'] = content[18]
+    rec['dc_format_s'] = content[19]
+    rec['dc_subject_sm'] = content[20]
+    rec['layer_id_s'] = content[21]
+    rec['layer_modified_dt'] = timestring
+    rec['layer_slug_s'] = content[23]
+    rec['layer_geom_type_s'] = content[24]
+
+    return rec.to_json
+  end
+
+
+  def ingestrec(solrurl, content)
```

Comment:
FWIW, the GeoBlacklight gem has a rake task that includes one to ingest JSON-encoded GeoHydra records into Solr.  (See [this rake task for details](https://github.com/geoblacklight/geoblacklight/blob/master/lib/tasks/geoblacklight.rake#L13).)

The heavy lifting in that rake task is done via a call to Blacklight.default_index.connection.add, which knows how to reach the Solr endpoint (and presumably also handles any other Blacklight side effects we might not know about at this stage).  This might avoid you having to mess around with trying to process the config/blacklight.yml file yourself to figure out where the core is that you want to use, which is what you do later in the run() method.

It may also be that Blacklight.default_index.connection.add might throw exceptions that we may be able to catch to do error checking/logging.  It may we worth looking into...